### PR TITLE
Adds some boring style basics

### PR DESCRIPTION
**Pull request by tildedave**

### CODE REVIEW

**Comment by tildedave** [comment]
ping

**Comment by tildedave** [comment]
https://github.com/Crowdtilt/tilt-web-style-guide/blob/boring-style-things/00-javascript.md

**Comment by tildedave** [comment]
ping

File: 00-javascript.md [hunk 1]
```diff
@@ -1,10 +1,110 @@
 # JavaScript
+
+## The Basics
+
+* 4 character space indents, no tabs
```
Comment (cohitre) [inline 00-javascript.md]: mention that we have `editorconfig` for this

File: 00-javascript.md [hunk 2]
```diff
@@ -1,10 +1,110 @@
...
+```
+
+Instead use ES6 arrow functions to automatically re-bind `this` in the scope of
+the anonymous function:
+
+```
+() => this.setState({ foo: 'bar' });
+```
+
+If the function you need to bind is a 1-line function you can just line the
```
Comment (cohitre) [inline 00-javascript.md]: worth mentioning that the line expression returns the value of the last run statement.

**Comment by cohitre** [comment]
lgtm.

**Pull request merged**